forem/forem
# Log prop-types warnings to console in development

## What type of PR is this? (check all applicable)


## Description

We use `PropTypes` throughout the Preact codebase to help make sure our components render as expected, but I noticed at some point we lost logging of these errors to the console - this PR reintroduces them.

After reintroducing this logging the console was pretty noisy (😅 ) with errors, and I've tidied up the main bulk of them here so devs aren't overloaded after this PR is merged. There are still a few errors peppered throughout (they will only show up in local development), but this would be a great area for first time contributors, and I'll raise a separate blanket issue for those fixes.

## Related Tickets & Documents

N/A

## QA Instructions, Screenshots, Recordings

- Run the app locally, and try passing in some bad props (or missing required props) to one of our Preact components. You should see an error in the console.

### UI accessibility concerns?

N/A

## Added/updated tests?

- [X] No, and this is why: not needed for linting tool

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Admin Guide](https://admin.forem.com/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
      or in a [forem.dev](http://forem.dev) post
      replace this line with details on why this change doesn't need to be
      shared_
